src/main/java/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,170 @@
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                order = getOrderFromResultSet(resultSet);
+                orders.add(order);
+            }
+            statement.close();
+            for (Order ord : orders) {
+                ord.setProducts(getProductsFromOrder(ord.getId(), connection));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get orders with user id: " + id, e);
+        }
+        return orders;
+    }
+
+    @Override
+    public Order create(Order order) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                            "INSERT INTO orders(user_id) VALUES (?)",
+                            Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, order.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                order.setId(resultSet.getLong(1));
+            }
+            statement.close();
+            return addProductsToOrder(order, connection);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create order", e);
+        }
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        Order order = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                            "SELECT * FROM orders WHERE deleted = false AND order_id = ?;");
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                order = getOrderFromResultSet(resultSet);
+            }
+            statement.close();
+            order.setProducts(getProductsFromOrder(order.getId(), connection));
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get order with id: " + id, e);
+        }
+        return Optional.ofNullable(order);
+    }
+
+    @Override
+    public List<Order> getAll() {
+        List<Order> orders = new ArrayList<>();
+        Order order;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                            "SELECT * FROM orders WHERE deleted = false;");
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                order = getOrderFromResultSet(resultSet);
+                statement.close();
```

Comment:
here you are closing the statement in the middle of working with the resultSet
does everything work correctly?

I would recommend to solve it the same way you did in `getUserOrders()`

Reply:
Fixed